dbusutil/dbusutil.go

```diff
@@ -0,0 +1,110 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package dbusutil
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/godbus/dbus"
+
+	"github.com/snapcore/snapd/dirs"
+)
+
+// isSessionBusLikelyPresent checks for the apparent availability of DBus session bus.
+//
+// The code matches what go-dbus does when it tries to detect the session bus:
+// - the presence of the environment variable DBUS_SESSION_BUS_ADDRESS
+// - the presence of the bus socket address in the file /run/user/UID/dbus-session
+// - the presence of the bus socket in /run/user/UID/bus
+func isSessionBusLikelyPresent() bool {
```

Comment:
I wonder if it would be useful to (apart from bool) return a hint about what clue was decisive (in case it's useful to show it if something else that assumes dbus fails later).

Reply:
Not at present, as we only use it exactly as in this branch, to ask go-dbus to open a connection. If we need more information we can certainly return it.